File: kotlin/PlanesAndroid/app/src/test/java/PlaneIntersectingPointIteratorTest.java
import org.junit.Test;
import com.google.common.truth.Truth;
import com.planes.common.Coordinate2D;
import com.planes.common.Plane;
import com.planes.common.PlaneIntersectingPointIterator;

import static com.google.common.truth.Truth.assertThat;

public class PlaneIntersectingPointIteratorTest {

    @Test
    public void PlaneIntersectingPointIterator_Values() {
        Coordinate2D point = new Coordinate2D(0, 0);
        PlaneIntersectingPointIterator pipi = new PlaneIntersectingPointIterator(point);

        int count = 0;

        while (pipi.hasNext()) {
            Plane pl = pipi.next();
            assertThat(pl.containsPoint(point)).isTrue();
            count++;
        }

        assertThat(count > 0).isTrue();
    }
}

File: kotlin/PlanesAndroid/app/src/test/java/PlaneIntersectingPointIteratorTest.kt
package com.planes.test
import com.google.common.truth.Truth
import com.planes.single_player_engine.Coordinate2D
import com.planes.single_player_engine.PlaneIntersectingPointIterator
import org.junit.Test

class PlaneIntersectingPointIteratorTest {
    @Test
    fun PlaneIntersectingPointIterator_Values() {
        val point = Coordinate2D(0, 0)
        val pipi = PlaneIntersectingPointIterator(point)
        var count = 0
        while (pipi.hasNext()) {
            val pl = pipi.next()
            Truth.assertThat(pl.containsPoint(point)).isTrue()
            count++
        }
        Truth.assertThat(count > 0).isTrue()
    }
}